### ISSUE TITLE
Upgrade "leveldb" package to version from upstream master

### ISSUE THREAD

**Issue by chfast** [ruslo/hunter#1781]
The upstream leveldb (https://github.com/google/leveldb) added Windows support recently.
Please merge recent upstream changes to https://github.com/hunter-packages/leveldb and tag new release.

**Comment by ruslo** [comment]
> Please merge recent upstream changes to https://github.com/hunter-packages/leveldb and tag new release

Latest release tag `1.20` is already in https://github.com/hunter-packages/leveldb You want latest master?

**Comment by chfast** [comment]
> You want latest master?

Yes.

**Comment by ruslo** [comment]
Done: https://github.com/hunter-packages/leveldb/tree/hunter-1.20-9ce3051

**Comment by chfast** [comment]
Does it require any additional changes to CMake to be compatible with Hunter?

I will try to add it to Hunter now.

**Comment by ruslo** [comment]
> Does it require any additional changes to CMake to be compatible with Hunter?

I don't know. If no additional changes needed then you can use upstream archive:
- https://github.com/google/leveldb/archive/9ce30510d482f5b2fa2965201453f0fc914f700c.zip

There is no need to maintain a fork.

**Comment by xsacha** [comment]
Yes, for some reason the last release was March 2017 despite a lot of progress since then.

It looks like it already supports exporting a target, with a namespace too so the fork probably isn't needed.

**Comment by ruslo** [comment]
> https://github.com/google/leveldb/pull/648

This change is not needed.

> https://github.com/google/leveldb/pull/647

Yes, that's something useful.

**Comment by xsacha** [comment]
Are you sure the include isn't needed?
I just tried to add it in to Hunter directly and none of the configs mentioned the include directory.

**Comment by ruslo** [comment]
> Are you sure the include isn't needed?

`CMAKE_INSTALL_INCLUDEDIR` is defined for all platforms. This configuration will work everywhere:
- https://cgold.readthedocs.io/en/latest/tutorials/install/layout.html

If it's not working, then the problem is about missing GNUInstallDirs or somewhere else, not in `CMAKE_INSTALL_INCLUDEDIR` variable.

**Comment by Invincibl-e** [comment]
> > Are you sure the include isn't needed?
> 
> `CMAKE_INSTALL_INCLUDEDIR` is defined for all platforms. This configuration will work everywhere:
> 
> * https://cgold.readthedocs.io/en/latest/tutorials/install/layout.html
> 
> If it's not working, then the problem is about missing GNUInstallDirs or somewhere else, not in `CMAKE_INSTALL_INCLUDEDIR` variable.

You mean need to add `include(GNUInstallDirs)` before the 
```CMake
target_include_directories(leveldb
  PUBLIC
    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
    $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
)
```
?

**Comment by ruslo** [comment]
> You mean need to add include(GNUInstallDirs) before the

@Invincibl-e Yes
